From 5b8e11c1075e8b8f95884e2779e30975ce87e8e3 Mon Sep 17 00:00:00 2001
From: JJ Roberts-White <computerfido@gmail.com>
Date: Fri, 21 Jan 2022 14:51:12 +1100
Subject: [PATCH] Add 'lemon' target to config.sub

---
 lib/config.sub | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/config.sub b/lib/config.sub
index d74fb6d..036b78c 100755
--- a/lib/config.sub
+++ b/lib/config.sub
@@ -1745,7 +1745,7 @@ case $os in
 	     | os2* | vos* | palmos* | uclinux* | nucleus* | morphos* \
 	     | scout* | superux* | sysv* | rtmk* | tpf* | windiss* \
 	     | powermax* | dnix* | nx6 | nx7 | sei* | dragonfly* \
-	     | skyos* | haiku* | rdos* | toppers* | drops* | es* \
+	     | skyos* | haiku* | rdos* | toppers* | drops* | es* | lemon* \
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
 	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | wasi* \
 	     | nsk* | powerunix* | genode* | zvmoe* | qnx* | emx* | zephyr*)
-- 
2.34.1

