diff --color -ruN Python-3.8.2/config.sub Python-3.8.2-bkup/config.sub
--- Python-3.8.2/config.sub	2020-02-25 08:36:25.000000000 +1100
+++ Python-3.8.2-bkup/config.sub	2020-09-04 21:02:20.906734393 +1000
@@ -1392,7 +1392,7 @@
 	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
 	      | -morphos* | -superux* | -rtmk* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
-	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* \
+	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* | -lemon* \
 	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -redox* | -bme* \
 	      | -midnightbsd*)
 	# Remember, each alternative MUST END IN *, to match a version number.
diff --color -ruN Python-3.8.2/configure.ac Python-3.8.2-bkup/configure.ac
--- Python-3.8.2/configure.ac	2020-02-25 08:36:25.000000000 +1100
+++ Python-3.8.2-bkup/configure.ac	2020-09-04 23:06:18.720274573 +1000
@@ -379,6 +379,9 @@
 	*-*-cygwin*)
 		ac_sys_system=Cygwin
 		;;
+	*-*-lemon*)
+		ac_sys_system=Lemon
+		;;
 	*-*-vxworks*)
 	    ac_sys_system=VxWorks
 	    ;;
@@ -431,6 +434,9 @@
 	*-*-vxworks*)
 		_host_cpu=$host_cpu
 		;;
+	*-*-lemon*)
+		_host_cpu=$host_cpu
+		;;
 	*)
 		# for now, limit cross builds to known configurations
 		MACHDEP="unknown"
@@ -2641,6 +2647,9 @@
 	CYGWIN*)
 		LDSHARED="gcc -shared -Wl,--enable-auto-image-base"
 		LDCXXSHARED="g++ -shared -Wl,--enable-auto-image-base";;
+    Lemon*)
+        LDSHARED='$(CC) -shared'
+        LDCXXSHARED='$(CXX) -shared';;
 	*)	LDSHARED="ld";;
 	esac
 fi
@@ -2676,6 +2685,9 @@
 		then CCSHARED="-fPIC"
 		else CCSHARED="-Kpic -belf"
 		fi;;
+    Lemon*)
+        CCSHARED="-fPIC"
+        ;;
 	VxWorks*)
 		CCSHARED="-fpic -D__SO_PICABILINUX__  -ftls-model=global-dynamic"
 	esac
@@ -2738,6 +2750,8 @@
 		LINKFORSHARED='-Wl,-E -N 2048K';;
 	VxWorks*)
 		LINKFORSHARED='--export-dynamic';;
+	Lemon*)
+		LINKFORSHARED='-export-dynamic';;
 	esac
 fi
 AC_MSG_RESULT($LINKFORSHARED)
