diff --color -ruN libressl-3.3.1/crypto/compat/arc4random.h lemon-libressl/crypto/compat/arc4random.h
--- libressl-3.3.1/crypto/compat/arc4random.h	2020-05-08 09:12:43.000000000 +1000
+++ lemon-libressl/crypto/compat/arc4random.h	2021-02-14 15:11:47.075497250 +1100
@@ -12,7 +12,7 @@
 #elif defined(__hpux)
 #include "arc4random_hpux.h"
 
-#elif defined(__linux__)
+#elif defined(__linux__) || defined(__lemon__)
 #include "arc4random_linux.h"
 
 #elif defined(__midipix__)
diff --color -ruN libressl-3.3.1/include/compat/machine/endian.h lemon-libressl/include/compat/machine/endian.h
--- libressl-3.3.1/include/compat/machine/endian.h	2020-12-09 04:32:48.000000000 +1100
+++ lemon-libressl/include/compat/machine/endian.h	2021-02-14 15:06:25.848420194 +1100
@@ -21,7 +21,7 @@
 #define BYTE_ORDER BIG_ENDIAN
 #endif
 
-#elif defined(__linux__) || defined(__midipix__)
+#elif defined(__linux__) || defined(__lemon__) || defined(__midipix__)
 #include <endian.h>
 
 #elif defined(__sun) || defined(_AIX) || defined(__hpux)
