diff --color -ruN lemon-freetype/builds/unix/config.sub freetype-2.10.4/builds/unix/config.sub
--- lemon-freetype/builds/unix/config.sub	2020-10-20 16:15:01.000000000 +1100
+++ freetype-2.10.4/builds/unix/config.sub	2020-10-23 12:58:23.258896191 +1100
@@ -1720,6 +1720,7 @@
 	     | skyos* | haiku* | rdos* | toppers* | drops* | es* \
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
 	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | wasi* \
+	     | lemon* \
 	     | nsk* | powerunix* | genode* | zvmoe* | qnx* | emx*)
 		;;
 	# This one is extra strict with allowed versions
diff --color -ruN lemon-freetype/builds/unix/configure freetype-2.10.4/builds/unix/configure
--- lemon-freetype/builds/unix/configure	2020-10-20 16:15:01.000000000 +1100
+++ freetype-2.10.4/builds/unix/configure	2020-10-23 12:59:44.788897447 +1100
@@ -4872,6 +4872,10 @@
   lt_cv_deplibs_check_method=pass_all
   ;;
 
+lemon*)
+  lt_cv_deplibs_check_method=pass_all
+  ;;
+
 hpux10.20* | hpux11*)
   lt_cv_file_magic_cmd=/usr/bin/file
   case $host_cpu in
@@ -10666,6 +10670,19 @@
   hardcode_into_libs=yes
   ;;
 
+lemon*)
+  version_type=linux
+  need_lib_prefix=no
+  need_version=no
+  dynamic_linker="$host_os ld"
+  library_names_spec='$libname$release$shared_ext$versuffix $libname$release$shared_ext$major $libname$shared_ext'
+  soname_spec='$libname$release$shared_ext$major'
+  shlibpath_var=LIBRARY_PATH
+  shlibpath_overrides_runpath=no
+  sys_lib_dlsearch_path_spec='/system/lib /lib'
+  hardcode_into_libs=yes
+  ;;
+
 hpux9* | hpux10* | hpux11*)
   # Give a soname corresponding to the major version so that dld.sl refuses to
   # link against other versions.
