diff --color -ruN curl-7.77.0/config.sub lemon-curl/config.sub
--- curl-7.77.0/config.sub	2021-05-18 17:07:51.000000000 +1000
+++ lemon-curl/config.sub	2021-07-05 23:57:25.208513051 +1000
@@ -1389,7 +1389,7 @@
 	      | -morphos* | -superux* | -rtmk* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
 	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* \
-	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -redox* | -bme* \
+	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -redox* | -bme* | -lemon* \
 	      | -midnightbsd*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
diff --color -ruN curl-7.77.0/lib/curl_setup.h lemon-curl/lib/curl_setup.h
--- curl-7.77.0/lib/curl_setup.h	2021-05-25 20:50:36.000000000 +1000
+++ lemon-curl/lib/curl_setup.h	2021-07-06 00:02:10.715170402 +1000
@@ -160,6 +160,7 @@
 /* ================================================================ */
 
 #include <curl/curl.h>
+#include <sys/select.h>
 
 /*
  * Disable other protocols when http is the only one desired.
