diff --color -ruN ncurses-6.2/config.sub lemon-ncurses/config.sub
--- ncurses-6.2/config.sub	2019-09-12 09:56:40.000000000 +1000
+++ lemon-ncurses/config.sub	2021-01-26 18:39:56.104399943 +1100
@@ -1363,7 +1363,7 @@
 	     | os2* | vos* | palmos* | uclinux* | nucleus* \
 	     | morphos* | superux* | rtmk* | windiss* \
 	     | powermax* | dnix* | nx6 | nx7 | sei* | dragonfly* \
-	     | skyos* | haiku* | rdos* | toppers* | drops* | es* \
+	     | skyos* | haiku* | rdos* | toppers* | drops* | es* | lemon* \
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
 	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | wasi* \
 	     | nsk* | powerunix)
diff --color -ruN ncurses-6.2/configure lemon-ncurses/configure
--- ncurses-6.2/configure	2020-02-09 08:06:19.000000000 +1100
+++ lemon-ncurses/configure	2021-01-26 18:44:23.223892256 +1100
@@ -6162,6 +6162,10 @@
 		fi
 		MK_SHARED_LIB='${LD} ${LDFLAGS} -d y -G -o $@'
 		;;
+  (lemon*)
+    CC_SHARED_OPTS="$CC_SHARED_OPTS -fPIC"
+    MK_SHARED_LIB='${CC} ${LDFLAGS} ${CFLAGS} -shared -Wl,-soname,'$cf_cv_shared_soname',-stats,-lc -o $@'
+    ;;
 	(*)
 		CC_SHARED_OPTS='unknown'
 		MK_SHARED_LIB='echo unknown'
Binary files ncurses-6.2/ncurses-6.2.tar.gz and lemon-ncurses/ncurses-6.2.tar.gz differ
