diff --color -Naur nasm-2.15.02/asm/nasm.c lemon-nasm/asm/nasm.c
--- nasm-2.15.02/asm/nasm.c	2020-07-17 20:27:24.276299338 +1000
+++ lemon-nasm/asm/nasm.c	2020-07-17 20:24:17.000000000 +1000
@@ -490,14 +490,15 @@
 
     best_gm = NULL;
 
-    tp = localtime(&oct->t);
+    struct tm faketp = {5, 5, 5, 5, 5, 100, 5, 5};
+    tp = &faketp;//localtime(&oct->t);
     if (tp) {
         oct->local = *tp;
         best_gm = &oct->local;
         oct->have_local = true;
     }
 
-    tp = gmtime(&oct->t);
+    //tp = gmtime(&oct->t);
     if (tp) {
         oct->gm = *tp;
         best_gm = &oct->gm;
diff --color -Naur nasm-2.15.02/autoconf/helpers/config.sub lemon-nasm/autoconf/helpers/config.sub
--- nasm-2.15.02/autoconf/helpers/config.sub	2020-07-17 20:27:24.276299338 +1000
+++ lemon-nasm/autoconf/helpers/config.sub	2020-07-17 20:24:17.000000000 +1000
@@ -1358,7 +1358,7 @@
 	     | os2* | vos* | palmos* | uclinux* | nucleus* \
 	     | morphos* | superux* | rtmk* | windiss* \
 	     | powermax* | dnix* | nx6 | nx7 | sei* | dragonfly* \
-	     | skyos* | haiku* | rdos* | toppers* | drops* | es* \
+	     | skyos* | haiku* | rdos* | toppers* | drops* | es* | lemon* \
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
 	     | midnightbsd*)
 	# Remember, each alternative MUST END IN *, to match a version number.
diff --color -Naur nasm-2.15.02/include/nasmlib.h lemon-nasm/include/nasmlib.h
--- nasm-2.15.02/include/nasmlib.h	2020-07-02 14:44:07.000000000 +1000
+++ lemon-nasm/include/nasmlib.h	2020-07-17 20:27:55.269123317 +1000
@@ -352,7 +352,7 @@
  */
 enum file_flags {
     NF_BINARY   = 0x00000000,   /* Binary file (default) */
-    NF_TEXT     = 0x00000001,   /* Text file */
+    NF_TEXT     = 0x00000000,   /* Text file */
     NF_NONFATAL = 0x00000000,   /* Don't die on open failure (default) */
     NF_FATAL    = 0x00000002,   /* Die on open failure */
     NF_FORMAP   = 0x00000004,   /* Intended to use nasm_map_file() */
diff --color -Naur nasm-2.15.02/output/outobj.c lemon-nasm/output/outobj.c
--- nasm-2.15.02/output/outobj.c	2020-07-17 20:27:24.276299338 +1000
+++ lemon-nasm/output/outobj.c	2020-07-17 20:24:17.000000000 +1000
@@ -1919,8 +1919,9 @@
 
     if (!nasm_file_time(&t, pathname))
         return 0;
-
-    lt = localtime(&t);
+        
+    struct tm fakelt = {5, 5, 5, 5, 5, 100, 5, 5};
+    lt = &fakelt;//localtime(&t);
     if (!lt)
         return 0;
 
