diff --color -ruN pixman-0.40.0/config.sub lemon-pixman/config.sub
--- pixman-0.40.0/config.sub	2020-04-20 07:53:21.000000000 +1000
+++ lemon-pixman/config.sub	2021-08-17 17:53:26.842662411 +1000
@@ -1384,7 +1384,7 @@
 	     | os2* | vos* | palmos* | uclinux* | nucleus* \
 	     | morphos* | superux* | rtmk* | windiss* \
 	     | powermax* | dnix* | nx6 | nx7 | sei* | dragonfly* \
-	     | skyos* | haiku* | rdos* | toppers* | drops* | es* \
+	     | skyos* | haiku* | rdos* | toppers* | drops* | es* | lemon* \
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
 	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | wasi* \
 	     | nsk* | powerunix)
