diff --color -ruN cairo-1.16.0/build/config.sub lemon-cairo/build/config.sub
--- cairo-1.16.0/build/config.sub	2018-08-17 11:11:52.000000000 +1000
+++ lemon-cairo/build/config.sub	2021-08-17 16:59:34.960533462 +1000
@@ -1388,7 +1388,7 @@
 	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
 	      | -morphos* | -superux* | -rtmk* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
-	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* \
+	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* | -lemon* \
 	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -redox* | -bme* \
 	      | -midnightbsd*)
 	# Remember, each alternative MUST END IN *, to match a version number.
