diff --color -ruN libedit-20191231-3.1/config.sub lemon-libedit/config.sub
--- libedit-20191231-3.1/config.sub	2019-12-31 21:58:29.000000000 +1100
+++ lemon-libedit/config.sub	2021-01-26 16:55:12.125792881 +1100
@@ -1390,7 +1390,7 @@
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
 	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* \
 	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -redox* | -bme* \
-	      | -midnightbsd*)
+	      | -midnightbsd* | -lemon* )
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
diff --color -ruN libedit-20191231-3.1/src/chartype.h lemon-libedit/src/chartype.h
--- libedit-20191231-3.1/src/chartype.h	2019-10-26 07:12:05.000000000 +1100
+++ lemon-libedit/src/chartype.h	2021-01-26 16:55:12.132459702 +1100
@@ -34,6 +34,7 @@
  * seems to actually advertise this properly, despite Unicode 3.1 having
  * been around since 2001... */
 #if	!defined(__NetBSD__) && \
+	!defined(__lemon__) && \
 	!defined(__sun) && \
 	!(defined(__APPLE__) && defined(__MACH__)) && \
 	!defined(__OpenBSD__) && \
diff --color -ruN libedit-20191231-3.1/src/editline/readline.h lemon-libedit/src/editline/readline.h
--- libedit-20191231-3.1/src/editline/readline.h	2019-10-26 07:12:05.000000000 +1100
+++ lemon-libedit/src/editline/readline.h	2021-01-26 16:55:12.139126520 +1100
@@ -76,7 +76,7 @@
 
 #ifndef CTRL
 #include <sys/ioctl.h>
-#if !defined(__sun) && !defined(__hpux) && !defined(_AIX)
+#if !defined(__sun) && !defined(__hpux) && !defined(_AIX) && !defined(__lemon__)
 #include <sys/ttydefaults.h>
 #endif
 #ifndef CTRL
diff --color -ruN libedit-20191231-3.1/src/vis.c lemon-libedit/src/vis.c
--- libedit-20191231-3.1/src/vis.c	2018-05-26 04:09:38.000000000 +1000
+++ lemon-libedit/src/vis.c	2021-01-26 16:55:12.142459930 +1100
@@ -319,7 +319,7 @@
 	/* See comment in istrsenvisx() output loop, below. */
 	wmsk = 0;
 	for (i = sizeof(wmsk) - 1; i >= 0; i--) {
-		shft = i * NBBY;
+		shft = i * CHAR_BIT;
 		bmsk = (uint64_t)0xffLL << shft;
 		wmsk |= bmsk;
 		if ((c & wmsk) || i == 0)
@@ -550,7 +550,7 @@
 			clen = 0;
 			wmsk = 0;
 			for (i = sizeof(wmsk) - 1; i >= 0; i--) {
-				shft = i * NBBY;
+				shft = i * CHAR_BIT;
 				bmsk = (uint64_t)0xffLL << shft;
 				wmsk |= bmsk;
 				if ((*dst & wmsk) || i == 0)
