From f15b084b9e3c3f6add76ec2216182bcfa7a54c6d Mon Sep 17 00:00:00 2001
From: JJ <computerfido@gmail.com>
Date: Fri, 5 Aug 2022 13:01:32 +1000
Subject: [PATCH] LemonOS specific changes

---
 config.sub | 2 +-
 configure  | 4 ++++
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/config.sub b/config.sub
index d74fb6d..036b78c 100755
--- a/config.sub
+++ b/config.sub
@@ -1745,7 +1745,7 @@ case $os in
 	     | os2* | vos* | palmos* | uclinux* | nucleus* | morphos* \
 	     | scout* | superux* | sysv* | rtmk* | tpf* | windiss* \
 	     | powermax* | dnix* | nx6 | nx7 | sei* | dragonfly* \
-	     | skyos* | haiku* | rdos* | toppers* | drops* | es* \
+	     | skyos* | haiku* | rdos* | toppers* | drops* | es* | lemon* \
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
 	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | wasi* \
 	     | nsk* | powerunix* | genode* | zvmoe* | qnx* | emx* | zephyr*)
diff --git a/configure b/configure
index 421cf85..39d77b7 100755
--- a/configure
+++ b/configure
@@ -6299,6 +6299,10 @@ echo "${ECHO_T}$cf_cv_ldflags_search_paths_first" >&6
 		fi
 		cf_cv_rm_so_locs=yes
 		;;
+	(lemon*)
+		CC_SHARED_OPTS='-fPIC'
+		MK_SHARED_LIB='${CC} ${LDFLAGS} ${CFLAGS} -shared -o $@'
+		;;
 	(linux*|gnu*|k*bsd*-gnu)
 		if test "$DFT_LWR_MODEL" = "shared" && test -n "$LD_RPATH_OPT" ; then
 			LOCAL_LDFLAGS="${LD_RPATH_OPT}\$(LOCAL_LIBDIR)"
-- 
2.37.1

