diff -ruN gcc-8.2.0/config.sub lemon-gcc/config.sub
--- gcc-8.2.0/config.sub	2018-01-03 15:25:18.000000000 +1100
+++ lemon-gcc/config.sub	2019-02-21 07:01:56.000000000 +1100
@@ -1416,7 +1416,7 @@
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
 	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* \
-	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -redox* | -bme*)
+	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -redox* | -bme* | -lemon*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
diff -ruN gcc-8.2.0/fixincludes/mkfixinc.sh lemon-gcc/fixincludes/mkfixinc.sh
--- gcc-8.2.0/fixincludes/mkfixinc.sh	2016-06-22 07:57:20.000000000 +1000
+++ lemon-gcc/fixincludes/mkfixinc.sh	2019-02-21 06:49:47.000000000 +1100
@@ -19,6 +19,7 @@
     powerpc-*-rtems*   | \
     powerpcle-*-eabisim* | \
     powerpcle-*-eabi* | \
+	*-lemon* | \
     *-musl* )
 	#  IF there is no include fixing,
 	#  THEN create a no-op fixer and exit
diff -ruN gcc-8.2.0/gcc/config/lemon.h lemon-gcc/gcc/config/lemon.h
--- gcc-8.2.0/gcc/config/lemon.h	1970-01-01 10:00:00.000000000 +1000
+++ lemon-gcc/gcc/config/lemon.h	2020-05-09 11:09:16.962892800 +1000
@@ -0,0 +1,38 @@
+/* Useful if you wish to make target-specific GCC changes. */
+#undef TARGET_LEMON
+#define TARGET_LEMON 1
+ 
+/* Default arguments you want when running your
+   i686-myos-gcc/x86_64-myos-gcc toolchain */
+#undef LIB_SPEC
+#define LIB_SPEC "-lc" /* link against C standard library */
+ 
+/* Files that are linked before user code.
+   The %s tells GCC to look for these files in the library directory. */
+#undef STARTFILE_SPEC
+#define STARTFILE_SPEC "%{!shared:crt0.o%s} crti.o%s %{shared:crtbeginS.o%s;:crtbegin.o%s}"
+ 
+/* Files that are linked after user code. */
+#undef ENDFILE_SPEC
+#define ENDFILE_SPEC "%{shared:crtendS.o%s;:crtend.o%s} crtn.o%s"
+
+#undef LINK_SPEC
+#define LINK_SPEC "%{shared:-shared} %{static:-static} %{!shared: %{!static: %{rdynamic:-export-dynamic}}}"
+
+#undef STANDARD_STARTFILE_PREFIX
+#define STANDARD_STARTFILE_PREFIX "/lib/"
+ 
+/* Don't automatically add extern "C" { } around header files. */
+#undef  NO_IMPLICIT_EXTERN_C
+#define NO_IMPLICIT_EXTERN_C 1
+ 
+/* Additional predefined macros. */
+#undef TARGET_OS_CPP_BUILTINS
+#define TARGET_OS_CPP_BUILTINS()      \
+  do {                                \
+    builtin_define ("__lemon__");      \
+    builtin_define ("__unix__");      \
+    builtin_assert ("system=lemon");   \
+    builtin_assert ("system=unix");   \
+    builtin_assert ("system=posix");   \
+  } while(0); 
diff -ruN gcc-8.2.0/gcc/config.gcc lemon-gcc/gcc/config.gcc
--- gcc-8.2.0/gcc/config.gcc	2018-06-26 05:34:01.000000000 +1000
+++ lemon-gcc/gcc/config.gcc	2020-05-09 11:15:07.593543600 +1000
@@ -744,6 +744,15 @@
   esac
   use_gcc_stdint=wrap
   ;;
+*-*-lemon*)
+	extra_options="$extra_options gnu-user.opt"
+	gas=yes
+	gnu_ld=yes
+	default_use_cxa_atexit=yes
+	use_gcc_stdint=wrap
+	tmake_file="${tmake_file} t-slibgcc"
+  	native_system_header_dir=/include
+	;;
 *-*-fuchsia*)
   native_system_header_dir=/include
   ;;
@@ -1542,6 +1551,9 @@
 	gas=yes
 	gnu_ld=yes
 	;;
+x86_64-*-lemon*)
+    tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h glibc-stdint.h i386/i386elf.h i386/x86-64.h gnu-user.h i386/gnu-user-common.h i386/gnu-user64.h lemon.h"
+    ;;
 i[34567]86-*-linux* | i[34567]86-*-kfreebsd*-gnu | i[34567]86-*-gnu* | i[34567]86-*-kopensolaris*-gnu)
 			# Intel 80386's running GNU/*
 			# with ELF format using glibc 2
diff -ruN gcc-8.2.0/libgcc/config.host lemon-gcc/libgcc/config.host
--- gcc-8.2.0/libgcc/config.host	2020-04-28 22:27:24.210769500 +1000
+++ lemon-gcc/libgcc/config.host	2018-04-07 06:04:17.000000000 +1000
@@ -641,14 +641,6 @@
 	;;
 x86_64-*-openbsd*)
 	;;
-i[34567]86-*-lemon*)
-	extra_parts="$extra_parts crti.o crtbegin.o crtend.o crtn.o"
-	tmake_file="$tmake_file i386/t-crtstuff t-crtstuff-pic t-libgcc-pic"
-	;;
-x86_64-*-lemon*)
-	extra_parts="$extra_parts crti.o crtbegin.o crtbeginS.o crtend.o crtendS.o crtn.o"
-	tmake_file="$tmake_file i386/t-crtstuff t-crtstuff-pic t-libgcc-pic t-slibgcc t-slibgcc-gld t-slibgcc-elf-ver t-libgcc-pic"
-	;;
 i[34567]86-*-linux*)
 	extra_parts="$extra_parts crtprec32.o crtprec64.o crtprec80.o crtfastmath.o"
 	tmake_file="${tmake_file} i386/t-crtpc t-crtfm i386/t-crtstuff t-dfprules"
diff -ruN gcc-8.2.0/libstdc++-v3/crossconfig.m4 lemon-gcc/libstdc++-v3/crossconfig.m4
--- gcc-8.2.0/libstdc++-v3/crossconfig.m4	2018-07-04 21:45:51.000000000 +1000
+++ lemon-gcc/libstdc++-v3/crossconfig.m4	2019-10-30 09:34:24.000000000 +1100
@@ -141,6 +141,13 @@
     AC_SUBST(SECTION_FLAGS)
     ;;
 
+  *-lemon*)
+    GLIBCXX_CHECK_COMPILER_FEATURES
+    GLIBCXX_CHECK_LINKER_FEATURES
+    GLIBCXX_CHECK_MATH_SUPPORT
+    GLIBCXX_CHECK_STDLIB_SUPPORT
+    ;;
+
   *-hpux*)
     SECTION_FLAGS='-ffunction-sections -fdata-sections'
     AC_SUBST(SECTION_FLAGS)
