diff -ruN binutils-2.32/bfd/config.bfd lemon-binutils/bfd/config.bfd
--- binutils-2.32/bfd/config.bfd	2019-01-20 03:01:32.000000000 +1100
+++ lemon-binutils/bfd/config.bfd	2020-05-13 20:10:57.309117500 +1000
@@ -727,6 +727,18 @@
     targ_defvec=i386_elf32_vec
     targ_selvecs=iamcu_elf32_vec
     ;;
+  i[3-7]86-*-lemon*)
+    targ_defvec=i386_elf32_vec
+    targ_selvecs=
+    targ64_selvecs=x86_64_elf64_vec
+    ;;
+#ifdef BFD64
+  x86_64-*-lemon*)
+    targ_defvec=x86_64_elf64_vec
+    targ_selvecs=i386_elf32_vec
+    want64=true
+    ;;
+#endif
   i[3-7]86-*-msdos*)
     targ_defvec=i386_aout_vec
     targ_selvecs=i386_msdos_vec
diff -ruN binutils-2.32/config.sub lemon-binutils/config.sub
--- binutils-2.32/config.sub	2019-01-20 03:01:33.000000000 +1100
+++ lemon-binutils/config.sub	2020-05-13 20:10:57.835782652 +1000
@@ -1363,7 +1363,7 @@
 	     | powermax* | dnix* | nx6 | nx7 | sei* | dragonfly* \
 	     | skyos* | haiku* | rdos* | toppers* | drops* | es* \
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
-	     | midnightbsd* | amdhsa* | unleashed* | emscripten*)
+	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | lemon*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	qnx*)
diff -ruN binutils-2.32/gas/configure.tgt lemon-binutils/gas/configure.tgt
--- binutils-2.32/gas/configure.tgt	2019-01-20 03:01:33.000000000 +1100
+++ lemon-binutils/gas/configure.tgt	2020-05-13 20:13:47.455294421 +1000
@@ -217,6 +217,7 @@
   i386-*-beos*)				fmt=elf ;;
   i386-*-elfiamcu)			fmt=elf arch=iamcu ;;
   i386-*-elf*)				fmt=elf ;;
+  i386-*-lemon*)			fmt=elf ;;
   i386-*-fuchsia*)			fmt=elf ;;
   i386-*-kaos*)				fmt=elf ;;
   i386-*-bsd*)				fmt=aout em=386bsd ;;
diff -ruN binutils-2.32/ld/configure.tgt lemon-binutils/ld/configure.tgt
--- binutils-2.32/ld/configure.tgt	2019-01-20 03:01:33.000000000 +1100
+++ lemon-binutils/ld/configure.tgt	2020-05-13 20:10:58.119115170 +1000
@@ -251,6 +251,10 @@
 			targ_extra_emuls="elf_iamcu"
 			targ64_extra_emuls="elf_x86_64 elf32_x86_64 elf_l1om elf_k1om"
 			targ64_extra_libpath="elf_x86_64 elf32_x86_64" ;;
+i[3-7]86-*-lemon*)	targ_emul=elf_i386_lemon
+			targ_extra_emuls="elf_i386 elf_x86_64_lemon elf_x86_64" ;;
+x86_64-*-lemon*)	targ_emul=elf_x86_64_lemon
+			targ_extra_emuls="elf_x86_64 elf_i386" ;; #elf_i386_lemon 
 x86_64-*-linux-gnux32)	targ_emul=elf32_x86_64
 			targ_extra_emuls="elf_x86_64 elf_i386 elf_iamcu elf_l1om elf_k1om"
 			targ_extra_libpath="elf_i386 elf_iamcu elf_x86_64 elf_l1om elf_k1om"
@@ -786,6 +790,10 @@
   NATIVE_LIB_DIRS='/lib /usr/lib /usr/pkg/lib /usr/local/lib'
   ;;
 
+*-*-lemon*)
+  NATIVE_LIB_DIRS='/lib /system/lib'
+  ;;
+
 *-*-freebsd*)
   NATIVE_LIB_DIRS='/lib /usr/lib /usr/local/lib'
   ;;
diff -ruN binutils-2.32/ld/emulparams/elf_i386_lemon.sh lemon-binutils/ld/emulparams/elf_i386_lemon.sh
--- binutils-2.32/ld/emulparams/elf_i386_lemon.sh	1970-01-01 10:00:00.000000000 +1000
+++ lemon-binutils/ld/emulparams/elf_i386_lemon.sh	2020-05-13 20:10:58.135781789 +1000
@@ -0,0 +1,3 @@
+. ${srcdir}/emulparams/elf_i386.sh
+GENERATE_SHLIB_SCRIPT=yes
+GENERATE_PIE_SCRIPT=yes
diff -ruN binutils-2.32/ld/emulparams/elf_x86_64_lemon.sh lemon-binutils/ld/emulparams/elf_x86_64_lemon.sh
--- binutils-2.32/ld/emulparams/elf_x86_64_lemon.sh	1970-01-01 10:00:00.000000000 +1000
+++ lemon-binutils/ld/emulparams/elf_x86_64_lemon.sh	2020-05-13 20:10:58.135781789 +1000
@@ -0,0 +1,3 @@
+. ${srcdir}/emulparams/elf_x86_64.sh
+GENERATE_SHLIB_SCRIPT=yes
+GENERATE_PIE_SCRIPT=yes
diff -ruN binutils-2.32/ld/Makefile.am lemon-binutils/ld/Makefile.am
--- binutils-2.32/ld/Makefile.am	2019-01-20 03:01:33.000000000 +1100
+++ lemon-binutils/ld/Makefile.am	2020-05-13 20:10:58.005782162 +1000
@@ -290,6 +290,7 @@
 	eelf_i386_nacl.c \
 	eelf_i386_sol2.c \
 	eelf_i386_vxworks.c \
+	eelf_i386_lemon.c \
 	eelf_iamcu.c \
 	eelf_s390.c \
 	eh8300elf.c \
@@ -454,6 +455,7 @@
 	eelf_k1om.c \
 	eelf_k1om_fbsd.c \
 	eelf_x86_64.c \
+	eelf_x86_64_lemon.c \
 	eelf_x86_64_cloudabi.c \
 	eelf_x86_64_fbsd.c \
 	eelf_x86_64_nacl.c \
@@ -1289,6 +1291,10 @@
 eelf_i386.c: $(srcdir)/emulparams/elf_i386.sh \
   $(ELF_X86_DEPS) $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
 
+eelf_i386_lemon.c: $(srcdir)/emulparams/elf_i386_lemon.sh \
+  $(ELF_DEPS) $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
+	${GENSCRIPTS} elf_i386_lemon "$(tdir_elf_i386_lemon)"
+
 eelf_i386_be.c: $(srcdir)/emulparams/elf_i386_be.sh \
   $(ELF_DEPS) $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
 
@@ -1849,6 +1855,10 @@
 eelf_x86_64.c: $(srcdir)/emulparams/elf_x86_64.sh \
   $(ELF_X86_DEPS) $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
 
+eelf_x86_64_lemon.c: $(srcdir)/emulparams/elf_x86_64_lemon.sh \
+	$(ELF_DEPS) $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
+		${GENSCRIPTS} elf_x86_64_lemon "$(tdir_elf_x86_64_lemon)"
+
 eelf_x86_64_cloudabi.c: $(srcdir)/emulparams/elf_x86_64_cloudabi.sh \
   $(srcdir)/emulparams/elf_x86_64.sh \
   $(ELF_X86_DEPS) $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
